plugins/outputs/postgresql/README.md

```diff
@@ -0,0 +1,194 @@
+| [float](https://docs.influxdata.com/influxdb/latest/reference/syntax/line-protocol/#float)                   | [double precision](https://www.postgresql.org/docs/current/datatype-numeric.html#DATATYPE-FLOAT)   |
+| [integer](https://docs.influxdata.com/influxdb/latest/reference/syntax/line-protocol/#integer)               | [bigint](https://www.postgresql.org/docs/current/datatype-numeric.html#DATATYPE-INT)               |
+| [uinteger](https://docs.influxdata.com/influxdb/latest/reference/syntax/line-protocol/#uinteger)             | [numeric](https://www.postgresql.org/docs/current/datatype-numeric.html#DATATYPE-NUMERIC-DECIMAL)* |
+| [string](https://docs.influxdata.com/influxdb/latest/reference/syntax/line-protocol/#string)                 | [text](https://www.postgresql.org/docs/current/datatype-character.html)                            |
+| [boolean](https://docs.influxdata.com/influxdb/latest/reference/syntax/line-protocol/#boolean)               | [boolean](https://www.postgresql.org/docs/current/datatype-boolean.html)                           |
+| [unix timestamp](https://docs.influxdata.com/influxdb/latest/reference/syntax/line-protocol/#unix-timestamp) | [timestamp](https://www.postgresql.org/docs/current/datatype-datetime.html)                        |
+
+It is important to note that `uinteger` (unsigned 64-bit integer) is mapped to the `numeric` PostgreSQL data type. The `numeric` data type is an arbitrary precision decimal data type that is less efficient than `bigint`. This is necessary as the range of values for the Influx `uinteger` data type can exceed `bigint`, and thus cause errors when inserting data.
+
+### pguint
+As a solution to the `uinteger`/`numeric` data type problem, there is a PostgreSQL extension that offers unsigned 64-bit integer support: https://github.com/petere/pguint.
+
+If this extension is installed, you can enable the `unsigned_integers` config parameter which will cause the plugin to use the `uint8` datatype instead of `numeric`.
+
+
+# Templating
+The postgresql plugin uses templates for the schema modification SQL statements. This allows for complete control of the schema by the user.
+
+Documentation on how to write templates can be found here:
+https://pkg.go.dev/github.com/influxdb/telegraf/plugins/outputs/postgresql/sqltemplate
+
+## Samples
+### TimescaleDB
+
+```toml
+tags_as_foreign_keys = true
+create_templates = [
+    '''CREATE TABLE {{.table}} ({{.columns}})''',
+    '''SELECT create_distributed_hypertable({{ .table|quoteLiteral }}, 'time', partitioning_column => 'tag_id', number_partitions => (SELECT count(*) FROM timescaledb_information.data_nodes)::integer, replication_factor => 2, chunk_time_interval => INTERVAL '1h')''',
+    '''ALTER TABLE {{ .table }} SET (timescaledb.compress, timescaledb.compress_segmentby = 'tag_id')''',
+]
+```
```

Comment:
Just tried this out and it works great 👍 
I had some trouble configuring it for a simple TimescaleDB setup as the README didn't have any example for that. Easy enough to sort out once I read up on the basics of Timescale, but it wouldn't hurt to start with a more basic setup here (create_hypertable instead of create_distributed_hypertable).

And here's a question: Is the any way of configuring this to use a separate table for each tag value using the templating? My use case is I'm using the mqtt_consumer and I'd like a separate table for each topic.

Reply:
Yeah, I hear what you're saying. The reason why I chose `create_distributed_hypertable` instead is because it's easier to take a more complex example and simplify it than it is to take a simple example and make it more complex. And creating examples for all possible configurations is not practical. It was my view that anyone using TimescaleDB would be familiar enough with the basics to understand and manipulate the provided example. Yes there is also the consideration that maybe one configuration is so popular that it should be documented, but all these considerations turn into a balancing game.

And yes, using a processor is how that scenario you mentioned should be handled.